extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketClient.java

```diff
@@ -275,14 +275,17 @@ public void sendNotification(Object data) {
         ExtensionServiceMessage msg = new ExtensionServiceMessage("");
         msg.fromMap(m);
         if (isConnected()) {
+            Semaphore localOutstandingNotifications = outstandingNotifications;
```

Comment:
you should just revert these changes completely.  Namir has fixed the problem as it was slightly more involved.  In the end, your PR should have no changes to this file.

Reply:
I have spent more than an hour already trying to revert. and yes, I already pulled the correction of Namir from master.
just forgive me on that one, any way it will merge with the master as it is a similar content.